### PR TITLE
Add new DocIndexRetriever example

### PR DESCRIPTION
**Pull request by xuechendi**
## Description

Add a new example for only add data to vector store and retrieve from vector store

related AIComps change:
https://github.com/opea-project/GenAIComps/pull/314

## Issues

`n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

No

## Tests

No

### CODE REVIEW

**Comment by xuechendi** [comment]
@minmin-intel @ftian1 @lvliang-intel , please take a look

**Review by minmin-intel — APPROVED** [review]
Looks good.

**Review by hshen14 — COMMENTED** [review]
@xuechendi It seems the new GenAI example could be part of ChatQnA (RAG). Does it make sense to improve ChatQnA to add this example?

**Comment by xuechendi** [comment]
> @xuechendi It seems the new GenAI example could be part of ChatQnA (RAG). Does it make sense to improve ChatQnA to add this example?

@hshen14 , thanks for the review.
The reason we extract RAG from ChatQnA is that we will not need LLM inference part. And we are planning to call this RAG megaservice by Agent.

If we use ChatQnA directly, we will need to provide different Spec from ChatQnA with different chatQnA megaflow to only enable RAG part.

Either way is working for me, but I felt adding a new example is clearer. Also, we can support more than one optional vectorstore backend in RAG megaservice for different scenarios.

**Comment by hshen14** [comment]
> > @xuechendi It seems the new GenAI example could be part of ChatQnA (RAG). Does it make sense to improve ChatQnA to add this example?
> 
> @hshen14 , thanks for the review. The reason we extract RAG from ChatQnA is that we will not need LLM inference part. And we are planning to call this RAG megaservice by Agent.
> 
> If we use ChatQnA directly, we will need to provide different Spec from ChatQnA with different chatQnA megaflow to only enable RAG part.
> 
> Either way is working for me, but I felt adding a new example is clearer. Also, we can support more than one optional vectorstore backend in RAG megaservice for different scenarios.

We are now having Agent example. Since it's designed to be used by Agent, how about moving under Agent?

**Review by XuhuiRen — APPROVED** [review]

File: DocIndexRetriever/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# DocRetriever Application
+
+DocRetriever are the most widely adopted use case for leveraging the different methodologies to match user query against a set of free-text records. DocRetriever is essential to RAG system, which bridges the knowledge gap by dynamically fetching relevant information from external sources, ensuring that responses generated remain factual and current. The core of this architecture are vector databases, which are instrumental in enabling efficient and semantic retrieval of information. These databases store data as vectors, allowing RAG to swiftly access the most pertinent documents or data points based on semantic similarity.
+
+## We provided DocWriterAgent with different deployment infra
```
Comment (minmin-intel) [inline DocIndexRetriever/README.md:5]: should be DocRetriever?

**Comment by xuechendi** [comment]
@kevinintel , can you help to '@' the right person who should review/approve this PR.
So we can provide it to RagAgent and InsightAgent who will treat DocRetriever as Tool.

**Pull request merged**